broker/query_plan.go

```diff
@@ -138,5 +222,43 @@ func (ap *AggQueryPlan) Run() (results queryCom.AQLQueryResult, err error) {
 //3. fan out requests, upon data from data nodes, flush to w
 //4. close, clean up, logging, metrics
 // TODO
-type NonAggQueryPlan struct {
+type NonAggQueryPlan struct{}
+
+// splitAvgQuery to sum and count queries
+func splitAvgQuery(q query.AQLQuery) (sumq query.AQLQuery, countq query.AQLQuery) {
```

Comment:
since we already support avg aggregation in a singlr query, not sure why you need to split

Reply:
because for each dimension, we need count from each data node to calculate the overall avg.
`totalAvt = (sum1 + sum2 +... + sumn) / (count1 + count2 + ... + countn)`

Reply:
should we consider break down avg and counts? because sometimes sum may cause overflow for some fields but we can still compute avg

Reply:
hmm data nodes can calculate avg even if sum would overflow?

Reply:
or even better: we can add a new agg function `rollingAvg`  at datanode level which returns avg and sum in 1 call.
let's keep it this way for now. will add todo for this one.

Reply:
sounds good for me. Sometimes we can compute avg without the total sum , that's what we are doing right now. 
Because for some column sum may not make sense but avg does.